lib/util/version.js

```diff
@@ -127,34 +136,45 @@ function getFlowVersionFromContext(context) {
     throw 'Could not retrieve flowVersion from settings'; // eslint-disable-line no-throw-literal
   }
   confVer = /^[0-9]+\.[0-9]+$/.test(confVer) ? `${confVer}.0` : confVer;
-  return confVer.split('.').map((part) => Number(part));
+  return normalizeParts(confVer.split('.').map(Number));
 }
 
-function normalizeParts(parts) {
-  return Array.from({length: 3}, (_, i) => (parts[i] || 0));
+function isLessThan(a, b) {
+  const higherMajor = a[0] < b[0];
+  if (higherMajor) {
+    return true;
+  }
+  const higherMinor = a[0] === b[0] && a[1] < b[1];
+  if (higherMinor) {
+    return true;
+  }
+  const higherOrEqualPatch = a[0] === b[0]
+    && a[1] === b[1]
+    && a[2] <= b[2];
+  return higherOrEqualPatch;
 }
```

Comment:
this logic doesn't seem like it will work for prerelease versions. i think we need to use the `semver` package for this.

Reply:
Interesting - it's equivalent (or should be - will double check this) to what was already here - note the removed lines from function `test` below.

Happy to consider using `semver`, but I was hoping to keep this PR functionally equivalent and just a pure performance boost - what do you think?

Reply:
I mean, we could add `semver` prior to this PR to correct the behavior, if you prefer :-p